usr/share/rear/rescue/default/500_ssh.sh

```diff
@@ -3,164 +3,97 @@
+# assume that we have openssh with configs in /etc/ssh
 
-    # The funny [] around a letter makes shopt -s nullglob remove this file from the list if it does not exist,
-    # files without a [] are mandatory.
+# The funny [] around a letter makes shopt -s nullglob remove this file from the list if it does not exist,
+# files without a [] are mandatory.
 
-    COPY_AS_IS=( "${COPY_AS_IS[@]}" /etc/ssh* /root/.ssh/[a]uthorized_keys /root/.shos[t]s )
-    PROGS=(
-        "${PROGS[@]}"
-        ssh sshd scp sftp ssh-agent
-        $(
-            read subsys sftp original_file junk < <( grep sftp /etc/sshd_co[n]fig /etc/ssh/sshd_co[n]fig /etc/openssh/sshd_co[n]fig /etc/centrifydc/ssh/sshd_co[n]fig 2>/dev/null )
-            echo $original_file
-        )
+COPY_AS_IS=( "${COPY_AS_IS[@]}" /etc/ssh* /root/.ssh/[a]uthorized_keys /root/.shos[t]s )
+PROGS=(
+    "${PROGS[@]}"
+    ssh sshd scp sftp ssh-agent
+    $(
+        read subsys sftp original_file junk < <( grep sftp /etc/sshd_co[n]fig /etc/ssh/sshd_co[n]fig /etc/openssh/sshd_co[n]fig /etc/centrifydc/ssh/sshd_co[n]fig 2>/dev/null )
+        echo $original_file
     )
-
-    # Part 1: SSH server (sshd) - this is for logging into the recovery system via SSH
-
-    # we need to add some specific NSS lib for shadow passwords to work on RHEL 6/7
-    LIBS=( "${LIBS[@]}" /usr/lib64/libfreeblpriv3.* /lib/libfreeblpriv3.* )
-    Log "Adding required libfreeblpriv3.so to LIBS"
-
-    # copy ssh user
-    # getent will return all entries that match the key(s) exactly - most systems use 'sshd', some may use 'ssh', none should use both.
-    # Only the first line (first returned entry) will be used by 'read' in 'IFS=: read ... <<<"$PASSWD_SSH"', so we ask for sshd first.
-    PASSWD_SSH=$(getent passwd sshd ssh)
-    if test -n "$PASSWD_SSH" ; then
-        # sshd:x:71:65:SSH daemon:/var/lib/sshd:/bin/false
-            IFS=: read user ex uid gid gecos homedir junk <<<"$PASSWD_SSH"
-        # skip if this user exists already in the restore system
-        if ! egrep -q "^$user:" $ROOTFS_DIR/etc/passwd ; then
-            echo "$PASSWD_SSH" >>$ROOTFS_DIR/etc/passwd
-        fi
-        # add ssh group to be collected later
-        CLONE_GROUPS=( "${CLONE_GROUPS[@]}" "$gid" )
-        mkdir -p $v -m 0700 "$ROOTFS_DIR/$homedir" >&2
-        chown $v root:root "$ROOTFS_DIR/$homedir" >&2
+)
+
+# Part 1: SSH server (sshd) - this is for logging into the recovery system via SSH
+
+# we need to add some specific NSS lib for shadow passwords to work on RHEL 6/7
+LIBS=( "${LIBS[@]}" /usr/lib64/libfreeblpriv3.* /lib/libfreeblpriv3.* )
+Log "Adding required libfreeblpriv3.so to LIBS"
+
+# copy ssh user
+# getent will return all entries that match the key(s) exactly - most systems use 'sshd', some may use 'ssh', none should use both.
+# Only the first line (first returned entry) will be used by 'read' in 'IFS=: read ... <<<"$PASSWD_SSH"', so we ask for sshd first.
+PASSWD_SSH=$(getent passwd sshd ssh)
```

Comment:
I still fail to see why we don't use `CLONE_USERS+( ssh sshd)` here instead. Then the only thing missing is the home directory, which should probably done in `build/default/500_patch_sshd_config.sh`